quickwit/quickwit-search/src/search_job_placer.rs

```diff
@@ -150,39 +150,51 @@ impl SearchJobPlacer {
         mut jobs: Vec<J>,
         excluded_addrs: &HashSet<SocketAddr>,
     ) -> anyhow::Result<impl Iterator<Item = (SearchServiceClient, Vec<J>)>> {
-        let num_nodes = self.searcher_pool.len();
+        let mut all_nodes = self.searcher_pool.pairs();
 
-        let mut candidate_nodes: Vec<CandidateNodes> = self
-            .searcher_pool
-            .pairs()
+        if all_nodes.is_empty() {
+            bail!(
+                "failed to assign search jobs: there are no available searcher nodes in the \
+                 cluster"
+            );
+        }
+        if !excluded_addrs.is_empty() && excluded_addrs.len() < all_nodes.len() {
```

Comment:
The condition `excluded_addrs.len() < all_nodes.len()` seems broken.
if there are more excluded addresses than nodes, we don't filter anything

The tests below don't cover `excluded_addrs`

Reply:
this seems to be on purpose. Per function doc:
> /// When exclude_addresses filters all clients it is ignored.